app/scripts/migrations/082.ts

```diff
@@ -27,12 +27,53 @@ export async function migrate(originalVersionedData: {
 function transformState(state: Record<string, unknown>) {
   if (
     !hasProperty(state, 'PreferencesController') ||
```

Comment:
I've recently discovered that it is possible for the preferences controller to not be set in state. It doesn't get set to anything unless a state change is made. So it shouldn't be an error if it's missing - it just means that the user has never changed the preferences state.

Reply:
See this snapshot for an example: https://github.com/MetaMask/metamask-extension/pull/20425

That snapshot is of the memstore not the persisted store. But the same should hold true of the persisted store since it's constructed in the same way (and the preferences controller is in both).

Reply:
addressed in e692373382